driver/multi.go

```diff
@@ -0,0 +1,143 @@
+package driver
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/brimsec/zq/ast"
+	"github.com/brimsec/zq/filter"
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/proc"
+	"github.com/brimsec/zq/scanner"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zng/resolver"
+	"go.uber.org/zap"
+)
+
+// A MultiSource is a set of one or more ZNG record sources, which could be
+// a zar archive, a local directory, a collection of remote objects, etc.
+//
+// If the sources are ordered, and that order is expected to be maintained
```

Comment:
> and that order is expected to be maintained on order-dependent query operations

Just going by the comment here (not the implementation), I'm not entirely sure I get why what OrderInfo returns should be affected by the query. In other words, it seems like `OrderInfo()`'s results are a property of the sources, irrespective of the query.